crates/oxc_linter/src/rules/no_bitwise.rs

```diff
@@ -0,0 +1,160 @@
+            allow: obj
+                .and_then(|v| v.get("allow"))
+                .and_then(serde_json::Value::as_array)
+                .map(|v| {
+                    v.iter()
+                        .filter_map(serde_json::Value::as_str)
+                        .map(ToString::to_string)
+                        .collect()
+                })
+                .unwrap_or_default(),
+            int32_hint: obj
+                .and_then(|v| v.get("int32Hint"))
+                .and_then(serde_json::Value::as_bool)
+                .unwrap_or_default(),
+        }
+    }
+
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        match node.get().kind() {
+            AstKind::BinaryExpression(bin_expr) => {
+                let op = bin_expr.operator.as_str();
+
+                if bin_expr.operator.is_bitwise()
+                    && !allowed_operator(&self.allow, op)
+                    && !is_int32_hint(self.int32_hint, node)
+                {
+                    ctx.diagnostic(NoBitwiseDiagnostic(op, bin_expr.span));
+                }
+            }
+            AstKind::UnaryExpression(unary_expr) => {
+                let op = unary_expr.operator.as_str();
+
+                if unary_expr.operator.is_bitwise()
+                    && !allowed_operator(&self.allow, op)
+                    && !is_int32_hint(self.int32_hint, node)
+                {
+                    ctx.diagnostic(NoBitwiseDiagnostic(op, unary_expr.span));
+                }
+            }
+            AstKind::AssignmentExpression(assign_expr) => {
+                let op = assign_expr.operator.as_str();
+
+                if assign_expr.operator.is_bitwise()
+                    && !allowed_operator(&self.allow, op)
+                    && !is_int32_hint(self.int32_hint, node)
+                {
+                    ctx.diagnostic(NoBitwiseDiagnostic(op, assign_expr.span));
+                }
+            }
+            _ => {}
+        }
+    }
+}
+
+fn allowed_operator(allow: &[String], operator: &str) -> bool {
+    allow.iter().any(|s| s == operator)
+}
+
+fn is_int32_hint(int32_hint: bool, node: &AstNode) -> bool {
+    if !int32_hint {
+        return false;
+    }
+
+    match node.get().kind() {
+        AstKind::BinaryExpression(bin_expr) => {
+            bin_expr.operator.as_str() == "|" && bin_expr.right.is_number_0()
+        }
+        AstKind::UnaryExpression(unary_expr) => {
+            unary_expr.operator.as_str() == "|" && unary_expr.argument.is_number_0()
+        }
+        AstKind::AssignmentExpression(assign_expr) => {
+            assign_expr.operator.as_str() == "|" && assign_expr.right.is_number_0()
+        }
+        _ => false,
+    }
+}
+
+#[test]
+fn test() {
+    use serde_json::json;
+
+    use crate::tester::Tester;
+
+    let pass = vec![
+        ("a + b", None),
+        ("!a", None),
+        ("a && b", None),
+        ("a || b", None),
+        ("a += b", None),
+        ("a &&= b", None),
```

Comment:
ESlint add [`parserOptions`](https://github.com/eslint/eslint/blob/e39f28d8578a00f4da8d4ddad559547950128a0d/tests/lib/rules/no-bitwise.js#L28-L30) for those, I not sure how we can add it, or does it needed for Oxc?